xwiki-commons-core/xwiki-commons-component/xwiki-commons-component-default/src/main/java/org/xwiki/component/embed/EmbeddableComponentManager.java

```diff
@@ -231,33 +233,43 @@ public <T> List<T> getInstanceList(Type role) throws ComponentLookupException
     @SuppressWarnings("unchecked")
     public <T> Map<String, T> getInstanceMap(Type roleType) throws ComponentLookupException
     {
-        Map<String, T> components = new HashMap<>();
-
-        Map<String, ComponentEntry<?>> entries = this.componentEntries.get(roleType);
-
-        // Add local components
-        if (entries != null) {
-            for (Map.Entry<String, ComponentEntry<?>> entry : entries.entrySet()) {
+        Map<String, ComponentEntry<?>> entries = new HashMap<>();
+        if (this.componentEntries.containsKey(roleType)) {
+            entries.putAll(this.componentEntries.get(roleType));
+            for (ComponentEntry<?> value : entries.values()) {
                 try {
-                    components.put(entry.getKey(), getComponentInstance((ComponentEntry<T>) entry.getValue()));
+                    getComponentInstance(value); // force loading the instance
```

Comment:
Not very happy about this, but not sure how to do better for now (see comment below)